src/dbnode/persist/fs/commitlog/commit_log_test.go

```diff
@@ -113,6 +113,12 @@ func newTestOptions(
 	return opts, scope
 }
 
+func randomByteArray(len int) ([]byte) {
```

Comment:
Nit: maybe better to just do a string; arbitrary bytes are harder to read in comparisons.

Reply:
Annotations are []byte and compared as such here - https://github.com/m3db/m3/blob/4842611956104b2643700ae7570f0a3cbbf2446a/src/dbnode/persist/fs/commitlog/commit_log_test.go#L168

I think if I use randomString instead of random byte array even then we will have to deal with hard to read byte array comparision.

Also generating a random string with requested number of bytes is hard because of the way characters are encoded. A character can get encoded to different number of bytes. So while its easy to generate a random string of desired number of characters its much harder to generate a random string of desired number of bytes. We will have to write a custom function to generate such a string. 

Due to these complexities, I am leaning towards using randomByteArray over random string.

Reply:
Fair points! Tbh what I usually do when comparing byte strings in tests is:

```
assert.Equal(t, string(w.a), string(annotation))
```
Makes the output generally much easier to read, while (afaik) still comparing the same data.